t/app/controller/dashboard.t

```diff
@@ -252,7 +259,30 @@ FixMyStreet::override_config {
         like $mech->res->header('Content-type'), qr'text/csv';
         $mech->content_contains('Report ID');
         $mech->delete_header('Authorization');
+
+        my $token = 'access_token=' . $counciluser->id . '-1234567890abcdefgh';
+        $mech->get_ok("/dashboard?export=2&$token");
+        is $mech->res->code, 202;
+        my $loc = $mech->res->header('Location');
+        like $loc, qr{/dashboard/csv/.*\.csv$};
+        $mech->get_ok("$loc?$token");
+        like $mech->res->header('Content-type'), qr'text/csv';
+        $mech->content_contains('Report ID');
     };
+
+    subtest 'view status page' => sub {
+        # Simulate a partly done file
+        my $f = Path::Tiny->tempfile(SUFFIX => '.csv-part', DIR => path($UPLOAD_DIR, 'dashboard_csv', $counciluser->id));
+        (my $name = $f->basename) =~ s/-part$//;;
+        $mech->log_in_ok( $counciluser->email );
+        $mech->get_ok('/dashboard/status');
+        $mech->content_contains('/dashboard/csv/www.example.org-body-' . $body->id . '-start_date-2014-01-02.csv');
+        $mech->content_like(qr/$name\s*<br>0KB\s*<i>In progress/);
+
+        my $token = 'access_token=' . $counciluser->id . '-1234567890abcdefgh';
+        $mech->get_ok("/dashboard/csv/$name?$token");
+        is $mech->res->code, 202;
+    }
```

Comment:
Should we not also check that if all files are generated it reflects that too. If for no other reason that to check we don't get endless request loops if someone leaves the tab open.